File: js/api/workouts.js
import { get, post, del } from './client.js';
import { ATHLETE_ID } from '../config.js';

export function saveWorkout(data) {
    return post('workouts', data);
}

export function getWorkouts(limit = 20) {
    return get('workouts', { athlete_id: ATHLETE_ID, limit });
}

export function getWorkoutDetail(id) {
    return get(`workouts/${id}`);
}

export function deleteWorkout(id) {
    return del(`workouts/${id}`);
}

File: server/routes/workouts.py
from fastapi import APIRouter, Depends

from server.dependencies import get_db
from server.routes.helpers import require_found
from server.models.workout import WorkoutSave
from server.services import workout_service

router = APIRouter()


@router.post("/workouts")
def save_workout(body: WorkoutSave, db=Depends(get_db)):
    return workout_service.save_workout(db, body)


@router.get("/workouts")
def get_workouts(athlete_id: int = 1, limit: int = 20, offset: int = 0, db=Depends(get_db)):
    return workout_service.get_workouts(db, athlete_id, limit, offset)


@router.get("/workouts/{id}")
def get_workout_detail(id: int, db=Depends(get_db)):
    return require_found(workout_service.get_workout_detail(db, id), "Workout")


@router.delete("/workouts/{id}")
def delete_workout(id: int, db=Depends(get_db)):
    return require_found(workout_service.delete_workout(db, id), "Workout")
